pkg/kubelet/kubelet.go

```diff
@@ -901,6 +901,23 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// If the cloud has zone information, label the node with the zone information
+		zones, ok := kl.cloud.Zones()
```

Comment:
@vishh What would be your motivation fro doing it from a helper pod rather than kubelet? (other than the basic decoupling argument)